### PR TITLE
Move to offix client

### PR DESCRIPTION
**Pull request by wongcmichelle**
[Notion ticket](https://www.notion.so/uwblueprintexecs/Replace-Apollo-Client-with-Offix-7728294826a64d6c8c453c3c2c178288)

## Changes
- Replace existing apollo client with offix client

## Testing
- Verify the following queries & mutations still work in the FE
Events
http://localhost:3000/events
- View all events
- Add
- Edit
- Archive
- Unarchive
- Delete

Team Members / Users
http://localhost:3000/manage/members
- View
- Add
- Edit
- Delete

Ambulances
http://localhost:3000/manage/ambulances
- View
- Add
- Edit
- Delete

Hospitals
http://localhost:3000/manage/hospitals
- View
- Add
- Edit
- Delete

Collection Points
http://localhost:3000/events/{eventId}
- View
- Add
- Edit
- Delete

Patients
http://localhost:3000/events/{eventId}/ccps/{ccpId}
- View
- Add
- Edit
- Delete

Location Pins
http://localhost:3000/events/{eventId}/map
- Add

before opening PR
- [x] check notion ticket
- [x] run linter
- [x] go through file diff

filling out PR
- [x] descriptive title
- [x] update `notion ticket` link
- [ ] update `figma link`
- [x] fill out `Changes`
  - [ ] \(optional) add inline comments in diff
- [x] fill out `Testing`
- [ ] \(optional) add `Screenshots`
- [x] assign yourself to the PR

after opening PR
- [x] link PR to notion ticket
- [x] move ticket to `In PR`
- [ ] make sure build passes
- [x] ping `@ps` in `#paramedics-dev`

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -19,93 +16,83 @@ import MapPage from './components/Maps/MapPage';
...
-            component={(props) => <PatientProfilePage mode="edit" {...props} />}
-          />
-          <Route exact path="/events/:eventId/map" component={MapPage} />
-          <Route path="/">
-            <Redirect to="/events" />
-          </Route>
-        </Switch>
-      </SnackbarProvider>
-    </ThemeProvider>
+    <Switch>
```
Comment (wongcmichelle) [inline src/App.tsx:20]: Basically just moving `ThemeProvider` and `SnackbarProvider` out to the parent component

File: src/components/Maps/SearchBar.tsx [hunk 2]
```diff
@@ -116,16 +116,17 @@ const SearchBar = ({
...
-                <ListItem
-                  key={suggestion.description}
-                  classes={{ root: styles.sidebarOptions }}
-                >
-                  <div {...getSuggestionItemProps(suggestion)}>
-                    <Typography>{suggestion.description}</Typography>
-                  </div>
-                </ListItem>
-              ))}
+              {!loading &&
```
Comment (wongcmichelle) [inline src/components/Maps/SearchBar.tsx]: This was a linter fix

**Comment by aaronabraham311** [comment]
Ran into this weird bug. I could previously do everything listed in testing except for CCP (I don't have the right user ID, but that's not your fault).  When I first went through the add event flow, the event was added but I saw these two warnings on the console:
Screen Shot 2020-11-25 at 3 11 18 PM
After a little bit more testing on various flows, I tried to add an event again but I wasn't able to as seen in the GIF.
Kapture 2020-11-25 at 15 27 34

**Comment by petorsong** [comment]
@aaronabraham311 the `missing field ambulances/hospitals` warning is unrelated to this PR and has already been flagged here
https://www.notion.so/uwblueprintexecs/frontend-errors-7f99d4607fc04483b7d49fc9a253f2b7

**Comment by wongcmichelle** [comment]
> Ran into this weird bug. I could previously do everything listed in testing except for CCP (I don't have the right user ID, but that's not your fault). When I first went through the add event flow, the event was added but I saw these two warnings on the console:
> Screen Shot 2020-11-25 at 3 11 18 PM
> After a little bit more testing on various flows, I tried to add an event again but I wasn't able to as seen in the GIF.
> Kapture 2020-11-25 at 15 27 34

@aaronabraham311 Is it possible that it's because the "location" shouldn't be a string of letters? Check the GraphQL response and see if you're getting an error.

File: src/graphql/apollo/client.ts [hunk 3]
```diff
@@ -1,11 +1,15 @@
...
-  clientState: {
-    defaults,
-  },
-});
+const config = {
+  link: new HttpLink({
+    uri: process.env.REACT_APP_BACKEND_HOST || 'http://localhost:4000/',
+  }),
+  cache: new InMemoryCache(),
+};
```
Comment (petorsong) [inline src/graphql/apollo/client.ts:10]: what are these updated config values doing?

File: src/graphql/apollo/defaults.ts [hunk 4]
```diff
@@ -1,10 +0,0 @@
-const defaults = {
-  events: {
-    name: '',
-  },
-  patients: {
-    gender: '',
-  },
-};
-
-export default defaults;
```
Comment (petorsong) [inline src/graphql/apollo/defaults.ts:10]: what was this doing before, do you know?

**Review by petorsong — DISMISSED** [review]
couldn't find anything that was broken, but would like some clarity with a couple of things

File: src/graphql/apollo/defaults.ts (hunk 4 above)
Comment (wongcmichelle) [reply src/graphql/apollo/defaults.ts:10]: If I recall correctly, this file wasn't being used. I think it used to be used for setting a "default" cache.

File: src/graphql/apollo/client.ts (hunk 3 above)
Comment (wongcmichelle) [reply src/graphql/apollo/client.ts:10]: They allow you to set up the client. https://offix.dev/docs/getting-started#using-the-client-inside-your-application

"apollo-cache-inmemory is the recommended cache implementation for Apollo Client 2.0. InMemoryCache is a normalized data store that supports all of Apollo Client 1.0's features without the dependency on Redux."

Comment (petorsong) [reply src/graphql/apollo/client.ts:10]: thanks! looks like we're using it in #63 as well so hopefully that makes things easier to merge and resolve conflicts

**Review by aaronabraham311 — DISMISSED** [review]

**Review by petorsong — APPROVED** [review]

**Review by aaronabraham311 — APPROVED** [review]

**Pull request merged**